src/mps/dmrg.jl

```diff
@@ -346,6 +347,7 @@ function dmrg(PH, psi0::MPS, sweeps::Sweeps; kwargs...)
         )
       end
     end
+    sweepup(typeof(data(psi[1])))
```

Comment:
Could you explain this? I'm guessing this is just a temporary thing but I don't think we always want to call this, at least on CPU.

Reply:
Of course: DMRG creates a lot of intermediate memory and will often get up to almost 100% of the GPU memory. I was trying to force GC and CUDA.reclaim to run to test and see if doing so influenced performance/memory usage. It has a very slight improvement in performance but I would think not necessary.

Reply:
Sounds good. For future reference, you can use an `Observer` to insert custom code into `dmrg` without modifying the source code of `dmrg`, see: https://itensor.discourse.group/t/memory-usage-in-dmrg-with-julia-1-x/1092.

Reply:
Good to know, thanks!